srk/src/interpretation.ml

```diff
@@ -389,7 +395,9 @@ let select_implicant interp ?(env=Env.empty) phi =
 
 let destruct_atom srk phi =
   match Formula.destruct srk phi with
-  | `Atom (op, s, t) -> `Comparison (op, s, t)
+  | `Atom (`Arith (op, s, t)) -> `Comparison (op, s, t)
+  | `Atom (`ArrEq _) ->  invalid_arg @@ Format.asprintf 
```

Comment:
I added this in. It is unclear to me if the changes I made to Quantifier.ml are correct. I more or less just copied the literal case to the "Arr_Comparison" case (except during testing if presburger), but I should really just be using "assert false" during the function related to mbp for the "Arr_Comparison" case.

Reply:
ArrEq is a better name than Arr_Comparison.  It's a particular type of comparison, and this project uses camel case for type constructors rather than underscores.